src/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonValueConverterEnum.cs

```diff
@@ -9,7 +9,17 @@ namespace System.Text.Json.Serialization.Converters
     internal sealed class JsonValueConverterEnum<TValue> : JsonValueConverter<TValue>
         where TValue : struct, Enum
     {
-        private static readonly bool s_isUint64 = Enum.GetUnderlyingType(typeof(TValue)) == typeof(ulong);
+        private static readonly bool s_isUInt64 = Enum.GetUnderlyingType(typeof(TValue)) == typeof(ulong);
```

Comment:
Setting all these statics isn't optimal. Can we do `Type.GetTypeCode(Enum.GetUnderlyingType(typeof(TValue))` and keep that as the static? Then you can switch on the TypeCode.

Reply:
https://github.com/dotnet/corefx/pull/38373#discussion_r292007211 cc: @jkotas 
just to understand, @JeremyKuhne is related to perf optimization of jit that elide switch branches if see readonly static or for style?Or because we'll do only one jump and not possible more than one if?

Reply:
You should measure. I do not think switch on typecode is going to be any faster in steady-state than this.

Reply:
> You should measure. I do not think switch on typecode is going to be any faster in steady-state than this.

The switch wasn't my concern, the initialization of a lot of statics was. I presume that Enum.GetUnderlyingType won't be combined into one call here. We're creating one of these for every enum which presumably would start to add up. Larger instance size as well.

Reply:
Ok thank's for explanation, I'll convert.

Reply:
whole type is going to be stamped for each enum. It would be certainly nice to make the stamp smaller.

You can measure that too: Create 10000 enums and serialize/deserialize each of them. Measure the time and footprint of the whole run (ie include startup). I typically measure just time because of it is easier and more stable; and it is proportional to footprint. Here is example of what I have done for a different PR earlier today: https://github.com/dotnet/coreclr/pull/24034#issuecomment-500656092